internal/experimental/experimental.go

```diff
@@ -0,0 +1,51 @@
+package experimental
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	envName = "NOTATION_EXPERIMENTAL"
+	enabled = "1"
+)
+
+// IsDisabled determines whether a experimental feature is disabled.
+func IsDisabled() bool {
+	return os.Getenv(envName) != enabled
+}
+
+// Error returns an error for a disabled experimental feature.
+func Error(description string) error {
+	return fmt.Errorf("%s is marked as experimental and not enabled by default. To use it, please set %s=%s in your environment", description, envName, enabled)
+}
+
+// CheckCommand checks whether a experimental command can be run.
+func CheckCommand(cmd *cobra.Command, args []string) error {
```

Comment:
Looks like this function only checks if a `command` is marked as experimental. How about flags? For example, under the `sign` command, there's the experimental flag `--signature-manifest`.

Reply:
This is a utility wrapper function of `func Check(cmd *cobra.Command, args []string, doCheck func(cmd *cobra.Command, args []string) (feature string, isExperimental bool)) error`, it checks whether a experimental command can be run.

If sign via an artifact manifest is experimental, you need to call `check` directly in sign command code like:
```golang
experimental.Check(cmd, args, func(cmd *cobra.Command, args []string) (string, bool)) {
    sig, err := cmd.Flags().GetString("signature-manifest")
    if err != nil || sig != "artifact" {
        // artifact not used
        return "", false
    }
    return "sign via OCI artifact", true
})
```